pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -33,6 +34,9 @@ type RecreateDeploymentStrategy struct {
 	until string
 	// getReplicationController knows how to get a replication controller.
 	getReplicationController func(namespace, name string) (*kapi.ReplicationController, error)
+	// listReplicationControllerEvents knows how to list all events related the
+	// given replication controller.
+	listReplicationControllerEvents func(rc *kapi.ReplicationController) (*kapi.EventList, error)
```

Comment:
Yes, but scope to the namespacers you need

On Wed, Sep 7, 2016 at 10:37 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/deploy/strategy/recreate/recreate.go
> <https://github.com/openshift/origin/pull/10816#discussion_r77780730>:
>
> > @@ -33,6 +34,9 @@ type RecreateDeploymentStrategy struct {
> >  	until string
> >  	// getReplicationController knows how to get a replication controller.
> >  	getReplicationController func(namespace, name string) (*kapi.ReplicationController, error)
> > +	// listReplicationControllerEvents knows how to list all events related the
> > +	// given replication controller.
> > +	listReplicationControllerEvents func(rc *kapi.ReplicationController) (*kapi.EventList, error)
>
> @kargakis <https://github.com/kargakis> sure, so the goal is to get rid
> of this methods and just use the client directly?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10816/files/2d8d2aaa56568d8fcdd6cadc2d19b792e38cc2aa#r77780730>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADuFf7nTb46CpVR3911Kri_j7AHAr1z7ks5qnnezgaJpZM4J11hc>
> .
>